### PR TITLE
DIRECTOR: LINGO: Add number of functions.

### PR DESCRIPTION
**Pull request by rvanlaar**
Implement the following Lingo functions:

numberofchars: the string length
numberofitems: count of ','. An empty string counts as 1 item.
numberoflines: count of '\n'. An empty string counts as 1 line.

### CODE REVIEW

File: engines/director/lingo/lingo-builtins.cpp [hunk 1]
```diff
@@ -1774,30 +1774,54 @@ void LB::b_window(int nargs) {
...
 
 void LB::b_numberofitems(int nargs) {
 	Datum d = g_lingo->pop();
 
-	warning("STUB: b_numberofitems");
-	d.toInt();
-	d.u.i = 0;
+	d.toString();
+	int numberofitems = 1;
+	Common::String contents = d.u.s->c_str();
```
Comment (sev-) [inline engines/director/lingo/lingo-builtins.cpp]: `Common::String contents = *d.u.s;` will avoid double object conversion.

File: engines/director/lingo/lingo-builtins.cpp [hunk 2]
```diff
@@ -1774,30 +1774,54 @@ void LB::b_window(int nargs) {
...
+	uint32 i = 0;
+	while (i < d.u.s->size()){
```
Comment (sev-) [inline engines/director/lingo/lingo-builtins.cpp]: Why not `for` here?

File: engines/director/lingo/lingo-builtins.cpp [hunk 3]
```diff
@@ -1774,30 +1774,54 @@ void LB::b_window(int nargs) {
...
 	Datum d = g_lingo->pop();
 
-	warning("STUB: b_numberoflines");
-	d.toInt();
-	d.u.i = 0;
+	d.toString();
+	int numberoflines = 1;
+	Common::String contents = d.u.s->c_str();
+	uint32 i = 0;
+	while (i < d.u.s->size()){
```
Comment (sev-) [inline engines/director/lingo/lingo-builtins.cpp]: please add space before '{'

File: engines/director/lingo/lingo-builtins.cpp (hunk 2 above)
Comment (rvanlaar) [reply engines/director/lingo/lingo-builtins.cpp]: No specific reason for this code. 
Mainly that other code in Scummvm that dealt with strings were using while loops.
I've converted both `while`s to `for`s.

File: engines/director/lingo/lingo-builtins.cpp (hunk 1 above)
Comment (rvanlaar) [reply engines/director/lingo/lingo-builtins.cpp]: Done

File: engines/director/lingo/lingo-builtins.cpp (hunk 3 above)
Comment (rvanlaar) [reply engines/director/lingo/lingo-builtins.cpp]: Done

**Comment by sev-** [comment]
Thanks, merging

**Pull request merged**